### PR TITLE
ARCH-4643 Add org-wide CODEOWNERS enforcement workflow

### PR DESCRIPTION
**Pull request by NielsDoucet**
### Description of your changes

Introduces an organisation-wide policy that disallows individual GitHub user accounts in CODEOWNERS files — only `@collibra/<team>` references (or no owner at all, to explicitly mark files as unowned) are permitted.

**What's added:**

- `org-workflows/enforce-codeowners-teams.yml` — GitHub Actions Required Workflow that runs on every PR across the entire org. Checks out the target repo and runs the validation script. Must be registered under *Org Settings → Actions → Required workflows* pointing to `collibra/.github/org-workflows/enforce-codeowners-teams.yml@main`.
- `scripts/check-codeowners.sh` — Shell script with two testable functions: `find_codeowners` (locates the CODEOWNERS file per GitHub's priority order) and `extract_individuals` (detects bare `@username` entries while ignoring teams, email addresses, comments, and pattern-only/no-owner lines).
- `scripts/check-codeowners.bats` — 16 BATS unit tests covering all edge cases.
- `.github/workflows/test-scripts.yml` — Repo-local workflow that runs the BATS suite on every push/PR touching `scripts/`.
- `README.md` — Documents the purpose and structure of `org-workflows/`, `.github/workflows/`, and `scripts/`.

---

### JIRA reference

ARCH-4643

---

### Impact Analysis

The Required Workflow will need to be registered by an org admin after merge. Until registered it has no effect on other repositories. The check is non-destructive: it only fails PRs that introduce new individual-user CODEOWNERS entries. Repositories that already have individual entries will surface failures only on their next PR.

---

#### Checklist

- [x] I have performed a self-review of my code
- [x] My code follows the contribution guidelines of this project
- [x] My changes generate no new warnings

### CODE REVIEW

File: .github/workflows/test-scripts.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+on:
+  push:
+    paths: ["scripts/**"]
+  pull_request:
+    paths: ["scripts/**"]
+
+jobs:
+  bats:
+    name: BATS
+    runs-on: ubuntu-latest
```
Comment (rathpc) [inline .github/workflows/test-scripts.yml]: we shouldn't be using `ubuntu-latest` but rather one of our runs-on runners

https://engineering-collibra.atlassian.net/wiki/spaces/CPL/pages/16339730650/RunsOn+-+AWS+Self+Hosted+Runners

File: .github/workflows/test-scripts.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - uses: actions/checkout@v4
```
Comment (rathpc) [inline .github/workflows/test-scripts.yml]: this action is out of date, we should use v6 now

File: org-workflows/enforce-codeowners-teams.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    branches: ["**"]
+
+jobs:
+  check-codeowners:
+    name: CODEOWNERS — teams only
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/checkout@v4
```
Comment (rathpc) [inline org-workflows/enforce-codeowners-teams.yml]: both of these should use v6

File: org-workflows/enforce-codeowners-teams.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+name: Enforce team-only CODEOWNERS
+
+on:
+  pull_request:
+    branches: ["**"]
+
+jobs:
+  check-codeowners:
+    name: CODEOWNERS — teams only
+    runs-on: ubuntu-latest
```
Comment (rathpc) [inline org-workflows/enforce-codeowners-teams.yml]: same as other workflow, should be using runs-on runners

https://engineering-collibra.atlassian.net/wiki/spaces/CPL/pages/16339730650/RunsOn+-+AWS+Self+Hosted+Runners

File: scripts/check-codeowners.sh
Comment (rathpc) [inline scripts/check-codeowners.sh:1]: more of a nit but this file is fairly inconsistent with the syntax it is using for conditionals. Can we standardize on just one? I am partial to always using double braces but I will leave it up to you

**Review by rathpc — CHANGES_REQUESTED** [review]

File: scripts/check-codeowners.sh
Comment (NielsDoucet) [reply scripts/check-codeowners.sh:1]: adjusted to use bash style consistently

File: .github/workflows/test-scripts.yml (hunk 1 above)
Comment (NielsDoucet) [reply .github/workflows/test-scripts.yml]: It looks like moving to the runner has made `bats` unavailable. I'll work with build engineering to get this sorted.

**Comment by NielsDoucet** [comment]
@rathpc could you take another look at this PR?

File: org-workflows/enforce-codeowners-teams.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    branches: ["**"]
+
+jobs:
+  check-codeowners:
+    name: CODEOWNERS — teams only
+    runs-on:
+      - runs-on
+      - runner=tn
+      - env=production-eu
+      - run-id=${{ github.run_id }}
```
Comment (rathpc) [inline org-workflows/enforce-codeowners-teams.yml]:
```suggestion
    runs-on:
      - runs-on=${{ github.run_id }}
      - runner=tn
      - env=production-eu
      - tag=check-codeowners
```

File: .github/workflows/test-scripts.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    paths: ["scripts/**"]
+
+jobs:
+  bats:
+    name: BATS
+    runs-on:
+      - runs-on
+      - runner=tn
+      - env=production-eu
+      - run-id=${{ github.run_id }}
```
Comment (rathpc) [inline .github/workflows/test-scripts.yml]:
```suggestion
    runs-on:
      - runs-on=${{ github.run_id }}
      - runner=tn
      - env=production-eu
      - tag=bats
```

**Review by rathpc — CHANGES_REQUESTED** [review]
two minor changes if you don't mind?

**Review by rathpc — APPROVED** [review]
